### PR TITLE
S32K1XX add dummy iwdg driver

### PR DESCRIPTION
**Pull request by PetervdPerk-NXP**
#19924 Needed this dummy iwdg driver since uavcannode needs this.

Since this patch da675372911b446460ab08b53b98c5c9f336520d

@davids5 what does this watchdog do, and is there a way to make this platform agnostic in the future?

### CODE REVIEW

**Comment by davids5** [comment]
> #19924 Needed this dummy iwdg driver since uavcannode needs this.
> 
> Since this patch [da67537](https://github.com/PX4/PX4-Autopilot/commit/da675372911b446460ab08b53b98c5c9f336520d)
> 
> @davids5 what does this watchdog do, and is there a way to make this platform agnostic in the future?

It is a watch dog that is enabled in the bootloader. It will reset the node if it is not pet in IIRC ~2 seconds. This insures bad code the CRC-ed ok can be replaced. 

> is there a way to make this platform agnostic in the future?

 it kind of is, in that is in arch and has generic names. What were you thinking we could do differently?

**Comment by PetervdPerk-NXP** [comment]
I thought the naming Independent watchdog (IWDG) was something special, but I guess renaming to wdg would be enough.

**Comment by davids5** [comment]
Yes - IIRC the STM32 have wdg, iwdg and wwdg. So it is up to the arch to chose what type of watch dog to use to meet the rather relaxed requirement allowing a boot (scripts etc ) of the IAVCAN application (after dispatch from the bootloader) to get to the loop in the node code to pet the dog.

File: platforms/nuttx/src/px4/nxp/s32k1xx/watchdog/iwdg.c [hunk 1]
```diff
@@ -0,0 +1,103 @@
...
+ *   none.
+ *
+ * Returned value:
+ *   none.
+ *
+ ****************************************************************************/
+
+void watchdog_init(void)
+{
+#if defined(CONFIG_STM32F7_JTAG_FULL_ENABLE) || \
```
Comment (davids5) [inline platforms/nuttx/src/px4/nxp/s32k1xx/watchdog/iwdg.c:80]: I would gut the comment out code and relay on a diff to help implement in future. Or add it now

**Review by davids5 — CHANGES_REQUESTED** [review]

**Review by davids5 — APPROVED** [review]
@PetervdPerk-NXP Looks good to me!

**Pull request merged**